### PR TITLE
docs(snowflake) Snowflake quick ingestion guide

### PR DESCRIPTION
**Pull request by maggiehays**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: docs/quick-ingestion-guides/snowflake/overview.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+* Tables
+* External Tables
+* Views
+* Materialized Views
+
+This recurring ingestion pipeline will also extract:
+
+* **Usage statistics** to help you understand recent query activity (available if using Snowflake Enterprise edition or above)
+* **Table- and Column-level lineage** to automatically define interdependencies between datasets and columns (available if using Snowflake Enterprise edition or above)
+* **Table- and column-level profile statistics** to help you understand the shape of the data
```
Comment (mayurinehate) [inline docs/quick-ingestion-guides/snowflake/overview.md:23]: Column level profiling is disabled by default. So probably change this to **Table-level profile statistics** ?

File: docs/quick-ingestion-guides/snowflake/overview.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+---
+title: Overview
+---
+# Snowflake Ingestion Guide: Overview
+
+## What You Will Get Out of This Guide
+
+This guide will help you set up the Snowflake connector through the DataHub UI to begin ingesting metadata into DataHub.
```
Comment (jjoyce0510) [inline docs/quick-ingestion-guides/snowflake/overview.md:8]: Let's drop any notion of the "UI". For most people, the UI will be synonymous with DataHub

File: docs/quick-ingestion-guides/snowflake/overview.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+Upon completing this guide, you will have a recurring ingestion pipeline that will extract metadata from Snowflake and load it into DataHub. This will include to following Snowflake asset types:
+
+* Databases
+* Schemas
+* Tables
+* External Tables
+* Views
+* Materialized Views
+
+This recurring ingestion pipeline will also extract:
```
Comment (jjoyce0510) [inline docs/quick-ingestion-guides/snowflake/overview.md:19]: Minor: The pipeline will also extract

File: docs/quick-ingestion-guides/snowflake/overview.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+You will NOT have extracted Stages, Snowpipes, Streams, Tasks, Procedures from Snowflake, as the connector does not support ingesting these assets yet.
+:::
+
+## Next Steps
+
+If that all sounds like what you're looking for, navigate to the [next page](setup.md), where we'll talk about prerequisites.
+
+## Advanced Guides and Reference
+
+If you're looking to do something more in-depth, want to use CLI instead of the DataHub UI, or just need to look at the reference documentation for this connector, use these links:
```
Comment (jjoyce0510) [inline docs/quick-ingestion-guides/snowflake/overview.md:35]: Can we avoid saying "use these links"?

Comment (jjoyce0510) [inline docs/quick-ingestion-guides/snowflake/overview.md:35]: "Something more in-depth" sounds very mysterious. Can we just say, If you want to ingest Metadata from snowflake using the DataHub CLI, check out the following resources:

File: docs/quick-ingestion-guides/snowflake/setup.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
+---
+title: Setup
+---
+# Snowflake Ingestion Guide: Setup & Prerequisites
+
+In order to configure ingestion from Snowflake, you'll first have to ensure you have a Snowflake user with the `ACCOUNTADMIN` role or `MANAGE GRANTS` privilege and you know where to [execute queries](https://docs.snowflake.com/en/user-guide/ui-worksheet.html#) in Snowflake.
```
Comment (jjoyce0510) [inline docs/quick-ingestion-guides/snowflake/setup.md:6]: This part is unclear to me:

"you know where to executed queries in Snowflake". I think this is unnecessary.

File: docs/quick-ingestion-guides/snowflake/setup.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+4. Assign privileges to extract lineage and usage statistics from Snowflake by executing the below query.
+
+   ```sql
+   grant imported privileges on database snowflake to role datahub_role;
+   ```
+
+## Next Steps
+
+Once you've done all of the above in Snowflake, it's time to [move on](configuration.md) to configuring the actual ingestion source within the DataHub UI
```
Comment (jjoyce0510) [inline docs/quick-ingestion-guides/snowflake/setup.md:63]: Let's avoid using "DataHub UI". DataHub Ui is the _default_ going forward.

For most people DataHub ====== DataHub ui.

Also, let's add punctuation

File: docs/quick-ingestion-guides/snowflake/setup.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
+---
+title: Setup
+---
+# Snowflake Ingestion Guide: Setup & Prerequisites
+
+In order to configure ingestion from Snowflake, you'll first have to ensure you have a Snowflake user with the `ACCOUNTADMIN` role or `MANAGE GRANTS` privilege and you know where to [execute queries](https://docs.snowflake.com/en/user-guide/ui-worksheet.html#) in Snowflake.
+
+## Snowflake Prerequisites
+
+1. Create a DataHub-specific role by executing the following queries. Replace `<your-warehouse>` with an existing warehouse that you wish to use for DataHub ingestion.
```
Comment (jjoyce0510) [inline docs/quick-ingestion-guides/snowflake/setup.md:10]: Minor: I had to assume a role first before these commands

```use role SYSADMIN;```

File: docs/quick-ingestion-guides/snowflake/setup.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+   ```sql
+   create user datahub_user display_name = 'DataHub' password='<your-password>' default_role = datahub_role default_warehouse = '<your-warehouse>';
+   -- Grant access to the DataHub role created above
+   grant role datahub_role to user datahub_user;
+   ```
+
+   Make note of the user and its password. You'll need this in the next step.
+
+3. Assign privileges to read metadata about your assets by executing the following queries. Replace `<your-database>` with an existing database. Repeat for all databases from your Snowflake instance that you wish to integrate with DataHub. This is a bit tedious but a very important step.
```
Comment (jjoyce0510) [inline docs/quick-ingestion-guides/snowflake/setup.md:30]: I think we can remove the commentary about things being tedious

**Comment by jjoyce0510** [comment]
Before merging, let's add the permissions required to extract profiles: 

```
// If you ARE using Snowflake Profiling feature: Grant select privileges to your tables and views 
grant select on all tables in database LONG_TAIL_COMPANIONS to role datahub_role_john;
grant select on future tables in database LONG_TAIL_COMPANIONS to role datahub_role_john;
grant select on all external tables in database LONG_TAIL_COMPANIONS to role datahub_role_john;
grant select on future external tables in database LONG_TAIL_COMPANIONS to role datahub_role_john;
grant select on all views in database LONG_TAIL_COMPANIONS to role datahub_role_john;
grant select on future views in database LONG_TAIL_COMPANIONS to role datahub_role_john;
```

**Comment by jjoyce0510** [comment]
All looks great to me! One thing I'd say -- Can we call out the caveats that I'd mentioned earlier?

1. By default, DataHub only profiles datasets that have changed in the past 1 day. This can be changed in the YAML editor by setting the value of `profile_if_updated_since_days` to something greater than 1. 
2. By default, DataHub only extracts usage and lineage information based on operations performed in the last 1 day. This can be changed by setting a custom value for `start_time` and `end_time` in the YAML editor. 

We can also say, "To learn more about setting these advanced values, check out the [Snowflake Recipe Guide.](https://datahubproject.io/docs/generated/ingestion/sources/snowflake/)."

**Review by jjoyce0510 — APPROVED** [review]
LGTM - Excited to get this in!

**Pull request merged**